plugins/outputs/zabbix/zabbix_test.go

```diff
@@ -0,0 +1,868 @@
+		Host:  "hostA",
+		Key:   "telegraf.lld.name.foo",
+		Value: `{"data":[{"{#FOO}":"bar"},{"{#FOO}":"moo"}]}`,
+		Clock: 0,
+	}
+
+	// Simulate a Zabbix server to get the data sent
+	listener, lerr := net.Listen("tcp", z.Address)
+	require.NoError(t, lerr)
+
+	defer listener.Close()
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	// Read first packet with two metrics, then the first autoregister packet and the second autoregister packet.
+	go func() {
+		// First packet with metrics
+		request := listenForZabbixMetric(t, listener, false)
+		compareData(t, []zabbixRequestData{zabbixMetric}, request.Data)
+
+		// Second packet, while time has not surpassed LLDSendInterval
+		request = listenForZabbixMetric(t, listener, false)
+		compareData(t, []zabbixRequestData{zabbixMetric}, request.Data)
+
+		// Third packet, time has surpassed LLDSendInterval, metrics + LLD
+		request = listenForZabbixMetric(t, listener, false)
+		if len(request.Data) == 2 {
+			request.Data[1].Clock = 0 // Ignore lld request clock
+			compareData(t, []zabbixRequestData{zabbixMetric, zabbixLLDMetric}, request.Data)
+		} else {
+			t.Errorf("Expected 2 metrics, got %d", len(request.Data))
+		}
+
+		// Fourth packet with metrics
+		request = listenForZabbixMetric(t, listener, false)
+		compareData(t, []zabbixRequestData{zabbixMetric}, request.Data)
+
+		// Fifth packet, time has surpassed LLDSendInterval, metrics. No LLD as there is nothing new.
+		request = listenForZabbixMetric(t, listener, false)
+		compareData(t, []zabbixRequestData{zabbixMetric}, request.Data)
+
+		// Sixth packet, new LLD info, but time has not surpassed LLDSendInterval
+		request = listenForZabbixMetric(t, listener, false)
+		compareData(t, []zabbixRequestData{zabbixMetricNew}, request.Data)
+
+		// Seventh packet, time has surpassed LLDSendInterval, metrics + LLD.
+		// Also, time has surpassed LLDClearInterval, so LLD is cleared.
+		request = listenForZabbixMetric(t, listener, false)
+		if len(request.Data) == 2 {
+			request.Data[1].Clock = 0 // Ignore lld request clock
+			compareData(t, []zabbixRequestData{zabbixMetric, zabbixLLDMetricNew}, request.Data)
+		} else {
+			t.Errorf("Expected 2 metrics, got %d", len(request.Data))
+		}
+
+		// Eighth packet, time host not surpassed LLDSendInterval, just metrics.
+		request = listenForZabbixMetric(t, listener, false)
+		compareData(t, []zabbixRequestData{zabbixMetric}, request.Data)
+
+		// Ninth packet, time has surpassed LLDSendInterval, metrics + LLD.
+		// Just the info of the zabbixMetric as zabbixMetricNew has not been seen since LLDClearInterval.
+		request = listenForZabbixMetric(t, listener, false)
+		if len(request.Data) == 2 {
+			request.Data[1].Clock = 0 // Ignore lld request clock
+			compareData(t, []zabbixRequestData{zabbixMetric, zabbixLLDMetric}, request.Data)
+		} else {
+			t.Errorf("Expected 2 metrics, got %d", len(request.Data))
+		}
+
+		wg.Done()
+	}()
+
+	// First packet
+	err = z.Write([]telegraf.Metric{m})
+	require.NoError(t, err)
```

Comment:
```suggestion
	require.NoError(t, z.Write([]telegraf.Metric{m}))
```

And the same for all following..